src/Autofac/Core/Lifetime/LifetimeScope.cs

```diff
@@ -101,10 +91,12 @@ protected LifetimeScope(IComponentRegistry componentRegistry, LifetimeScope pare
         /// <param name="componentRegistry">Components used in the scope.</param>
         public LifetimeScope(IComponentRegistry componentRegistry, object tag)
         {
-            _sharedInstances[SelfRegistrationId] = this;
             ComponentRegistry = componentRegistry ?? throw new ArgumentNullException(nameof(componentRegistry));
             Tag = tag ?? throw new ArgumentNullException(nameof(tag));
+
+            _sharedInstances[SelfRegistrationId] = this;
             RootLifetimeScope = this;
+            this.DiagnosticSource = new DiagnosticListener("Autofac");
```

Comment:
Remaining `this.`

Reply:
Removed all `this.` from the src/** set of files.